core/shared/src/main/scala/org/http4s/multipart/MultipartDecoder.scala

```diff
@@ -138,9 +138,8 @@ private[http4s] object MultipartDecoder {
               .map[Either[DecodeFailure, Multipart[F]]](parts =>
                 Right(Multipart(parts, Boundary(boundary)))
               )
-              .handleError {
-                case e: InvalidMessageBodyFailure => Left(e)
-                case e => Left(InvalidMessageBodyFailure("Invalid multipart body", Some(e)))
+              .recover { case e: DecodeFailure =>
```

Comment:
I think `MessageBodyFailure` is the right level of granularity. `DecodeFailure` is too general e.g. applies to headers.

```suggestion
              .recover { case e: MessageBodyFailure =>
```